pkg/api/store/workload/transform_store.go

```diff
@@ -61,3 +63,58 @@ func SetPublicEnpointsFields(data map[string]interface{}) {
 		}
 	}
 }
+
+func setTransitioning(apiContext *types.APIContext, data map[string]interface{}) {
+	if data["transitioning"] == "yes" || data["transitioning"] == "error" {
+		return
+	}
+	workloadType := convert.ToString(values.GetValueN(data, "type"))
+	switch workloadType {
+	case "/v3/project/schemas/deployment":
+		setTransitioningForDeployment(apiContext, data, "replicaSetStatus", "replicas", "readyReplicas")
+	case "/v3/project/schemas/daemonSet":
+		setTransitioningForDaemonSet(apiContext, data, "daemonSetStatus", "desiredNumberScheduled", "numberReady")
+	}
+}
+
+func setTransitioningForDeployment(apiContext *types.APIContext, data map[string]interface{}, statusField string, desiredField string, currField string) {
+	deploymentRevisionID := convert.ToString(values.GetValueN(data, "workloadAnnotations", "deployment.kubernetes.io/revision"))
+	if deploymentRevisionID == "1" {
+		return
+	}
+	namespace := convert.ToString(data["namespaceId"])
+	name := convert.ToString(data["name"])
+	replicaSets := customizeWorkload.GetRevisions(apiContext, namespace, name, "")
```

Comment:
this call might get too expensive given it is done on every deployment list. We fetch all replica sets for the namespace. Can we at list limit fetching the replica sets for this particular deployment? Can we also look more on the deployment to see if there is at least one state indicating that its replica sets are still going through the upgrade. 
Another possible solution - see if we can create a small controller setting Upgrading state somewhere in the deployment annotations when the replica set is upgrading.